backend/src/ml_space_lambda/config_profiles/lambda_functions.py

```diff
@@ -0,0 +1,135 @@
+import json
+import logging
+import time
+
+from ml_space_lambda.data_access_objects.config_profiles import ConfigProfilesDAO, ConfigProfileModel
+from ml_space_lambda.utils.common_functions import api_wrapper, generate_exception_response, ApiResponse
+from ml_space_lambda.utils.mlspace_config import get_environment_variables
+
+log = logging.getLogger(__name__)
+
+# DAO for DynamoDB operations
+config_profiles_dao = ConfigProfilesDAO()
+# Environment flags
+env_vars = get_environment_variables()
+
+@api_wrapper
+def list_profiles(event, context):
+    """
+    GET /config-profiles
+    Returns list of all profiles (metadata only).
+    """
+    try:
+        items = config_profiles_dao.list()
+        return ApiResponse(
+            body=items,
+            status_code=200
+        )
+    except Exception as e:
+        log.exception("Failed to list config profiles")
+        raise e
+
+@api_wrapper
+def get_profile(event, context):
+    """
+    GET /config-profiles/{profileId}
+    Returns full profile record.
+    """
+    profile_id = event["pathParameters"]["profileId"]
+    try:
+        item = config_profiles_dao.get(profile_id)
+        if not item:
+            return ApiResponse(
+                body={"message": "Profile not found"},
+                status_code=404
+            )
+        return ApiResponse(
+            body=item.to_dict(),
+            status_code=200
+        )
+    except Exception as e:
+        log.exception(f"Failed to retrieve profile {profile_id}")
+        raise e
+
+@api_wrapper
+def create_profile(event, context):
+    """
+    POST /config-profiles
+    Creates a new configuration profile.
+    """
+    try:
+        payload = json.loads(event["body"])
+        # Build model and add audit fields
+        now = int(time.time())
+        principal_id = event["requestContext"]["authorizer"]["principalId"]
+        profile = ConfigProfileModel(
+            name=payload["name"],
+            description=payload.get("description"),
+            notebook_instance_types=payload["notebookInstanceTypes"],
+            training_job_instance_types=payload["trainingJobInstanceTypes"],
+            hpo_job_instance_types=payload["hpoJobInstanceTypes"],
+            transform_job_instance_types=payload["transformJobInstanceTypes"],
+            endpoint_instance_types=payload["endpointInstanceTypes"],
+            profile_id=None,  # DAO will assign
+            created_by=principal_id,
+            created_at=now,
+            updated_by=principal_id,
+            updated_at=now,
+        )
```

Comment:
Could simplify this a lot by doing something like this…

```suggestion
        payload = json.loads(event["body"])
        # Build model and add audit fields
        principal_id = event["requestContext"]["authorizer"]["principalId"]
        payload["created_by"] = principal_id
        payload["updated_by"] = principal_id
        now = int(time.time())
        payload["created_at"] = now
        payload["updated_at"] = now
        profile = ConfigProfileModel(**payload)
```